src/cdk-experimental/popover-edit/edit-events.ts

```diff
@@ -0,0 +1,75 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Subject, timer} from 'rxjs';
+import {audit, distinctUntilChanged, filter, map, share} from 'rxjs/operators';
+import {Injectable} from '@angular/core';
+import {closest, CELL_SELECTOR, DEFAULT_HOVER_DELAY_MS, ROW_SELECTOR} from './common';
+
+/**
+ * Service for sharing delegated events and state for triggering edits.
+ */
+@Injectable()
+export class EditEvents {
+  readonly editing = new Subject<Element|null>();
+  readonly hovering = new Subject<Element|null>();
+  readonly mouseMove = new Subject<Element|null>();
+
+  protected currentlyEditing: Element|null = null;
+
+  protected readonly hoveringDistinct = this.hovering.pipe(distinctUntilChanged(), share());
+  protected readonly editingDistinct = this.editing.pipe(distinctUntilChanged(), share());
+
+  constructor() {
+    this.editing.subscribe(cell => {
+      this.currentlyEditing = cell;
+    });
+  }
+
+  /**
+   * Returns an Observable that emits true when the specified element's cell
+   * is editing and false when not.
+   */
+  editingCell(element: Element|EventTarget) {
+    let cell: Element|null = null;
+
+    return this.editing.pipe(
+        map(editCell => editCell === (cell || (cell = closest(element, CELL_SELECTOR)))),
```

Comment:
It seems the caller of the function should be responsible for sending in the right element, rather than it traversing to find the ancestor cell.

Reply:
Doing it this way allows us to evaluate it lazily (once the first editing event comes through) rather than greedily at startup.